computation_graph/graph_runners.py

```diff
@@ -77,13 +78,22 @@ def unary_with_state_and_expectations(
 def variadic_with_state_and_expectations(g, sink):
     f = run.to_callable_strict(g)
 
-    def inner(turns):
-        prev = {}
-        for turn, expectation in turns:
-            prev = f(prev, turn)
-            assert (
-                prev[graph.make_computation_node(sink)] == expectation
-            ), f"actual={prev[graph.make_computation_node(sink)]}\n expected: {expectation}"
+    if asyncio.iscoroutinefunction(f):
+        async def inner(turns):
+            prev = {}
+            for turn, expectation in turns:
+                prev = await f(prev, turn)
+                assert (
+                    prev[graph.make_computation_node(sink)] == expectation
+                ), f"actual={prev[graph.make_computation_node(sink)]}\n expected: {expectation}"
+    else:
```

Comment:
please return before this line inner, and then you don't need the else which would be more readable